### DISCUSSION TITLE
why does `--dry-run` not `prepare` ?

### DISCUSSION THREAD

**Discussion by charlesthomas** [semantic-release/semantic-release#2580] [Release workflow]
in our CI we want to always build a push to any branch to find out as soon as possible whether it broke the build. we're already running `semantic-release --dry-run` but that doesn't `prepare` which means our plugins for things like building Docker images don't run.

because of that we have to support two different ways of doing all of our build automation. one inside `semantic-release` / a plugin and one outside of it.

what is the reasoning behind not running `prepare` in `--dry-run` mode? would a PR to make it do that be accepted? what about an extra flag, if running `prepare` in `--dry-run` would violate a design principal?

**Comment by esatterwhite** [comment]
I agree it should do this. I think there also needs to be a generic `done` / `complete` hook to clean up when done. currently you would have to bake the same logic into both the `success` and `fail` hooks.
